sdk/cognitivelanguage/azure-ai-language-conversations/samples/async/authoring/sample_assign_deployment_resources.py

```diff
@@ -0,0 +1,100 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+FILE: sample_assign_deployment_resources.py
+
+DESCRIPTION:
+    This sample demonstrates how to assign deployment resources to a project and deploy a project to specific deployment resources. Assigning resources allow you to train your model in one resource and deploy them to other assigned resources.
+
+USAGE:
+    python sample_assign_deployment_resources.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_CONVERSATIONS_ENDPOINT             - endpoint for your CLU resource.
+    2) AZURE_CLIENT_ID                          - the client ID of your active directory application.
+    3) AZURE_TENANT_ID                          - the tenant ID of your active directory application.
+    4) AZURE_CLIENT_SECRET                      - the secret of your active directory application.
+"""
+
+"""
+Assigning deployment resources requires you to authenticate with AAD, you cannot assign deployment resources with key based authentication. You must have the role Cognitive Services Language Owner assigned to the authoring resource and the target resources.
+"""
+
+import asyncio
+
+async def sample_assign_resources():
+    import os
+    from azure.identity import DefaultAzureCredential
+    from azure.ai.language.conversations.authoring import ConversationAuthoringClient
+
+    clu_endpoint = os.environ["AZURE_CONVERSATIONS_ENDPOINT"]
+    
+    client = ConversationAuthoringClient(
+        clu_endpoint, credential=credential
+    )
+    
+    project_name = "test_project"
+
+    credential = DefaultAzureCredential()
+
+    poller = await client.begin_assign_resource(
+        project_name=project_name,
+        resourcesMetadata: [
+          {
+            "azureResourceId": "/subscriptions/80000000-0000-4e8a-0000-7ce285849735/resourceGroups/test-rg/providers/Microsoft.CognitiveServices/accounts/LangTestWeu",
+            "customDomain": "lang-test-weu.cognitiveservices.azure.com",
+            "region": "westeurope"
+          },
+          {
+            "azureResourceId": "/subscriptions/80000000-0000-4e8a-0000-7ce285849735/resourceGroups/test-rg/providers/Microsoft.CognitiveServices/accounts/LangTestEus",
+            "customDomain": "lang-test-eus.cognitiveservices.azure.com",
+            "region": "eastus"
+          }
+        ],
+    )
+    
+    response = await poller.result()
+    print(response)
+
+async def sample_deploy_model():
+    import os
+    from azure.core.credentials import AzureKeyCredential
+    from azure.ai.language.conversations.authoring import ConversationAuthoringClient

```

Comment:
if we're already authenticating with AAD, we can probably just use the same kind of credential for this instead of switching to API key